File: guides/micronaut-security-jwt/kotlin/src/main/kotlin/example/micronaut/HomeController.kt
package example.micronaut

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import io.micronaut.security.annotation.Secured
import java.security.Principal

@Secured("isAuthenticated()") // <1>
@Controller("/") // <2>
class HomeController {

    @Produces(MediaType.TEXT_PLAIN) // <3>
    @Get("/")  // <4>
    fun index(principal: Principal): String {  // <5>
        return principal.name
    }
}

File: guides/micronaut-security-basicauth/python/src/example/micronaut/home_controller.py
from java.security import Principal
from micronaut.http import MediaType
from micronaut.http.annotation import Get, Produces
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule


# tag::clazz[]
@Produces(MediaType.TEXT_PLAIN)
@Secured(SecurityRule.IS_AUTHENTICATED)  # <1>
@Get("/")  # <2> <3>
def index(principal: Principal) -> str:  # <4>
    return principal.getName()
# end::clazz[]
